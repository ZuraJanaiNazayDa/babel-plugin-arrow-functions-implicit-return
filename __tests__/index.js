const { transform } = require("babel-core");
const arrowImplicitReturn = require("../lib/index").default;

console.log(arrowImplicitReturn);

describe("transform arrow functions", () => {
    test("works like do expressions", () => {
        const source = `
            const fn = (x) => {
                if(x>0){
                    "x>0";
                } else {
                    "x<=0";
                }
            };
        `;
        const { code } = transform(source, { plugins: [arrowImplicitReturn] });
        expect(code).toMatchSnapshot();
    });

    test("returns last directive if body is empty", () => {
        const source = `
            const fn = (x) => {
                "foo";
                "bar";
            };
        `;
        const { code } = transform(source, { plugins: [arrowImplicitReturn] });
        expect(code).toMatchSnapshot();
    });

    test("empty block returns undefined", () => {
        const source = `
            const fn = (x) => {};
        `;
        const { code } = transform(source, { plugins: [arrowImplicitReturn] });
        expect(code).toMatchSnapshot();
    });
    test("expressions transforms correctly", () => {
        const source = `
            const fn = (x) => x+1;
        `;
        const { code } = transform(source, { plugins: [arrowImplicitReturn] });
        expect(code).toMatchSnapshot();
    });
});
