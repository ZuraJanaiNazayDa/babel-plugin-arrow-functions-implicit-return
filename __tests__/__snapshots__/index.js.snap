// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform arrow functions empty block returns undefined 1`] = `
"
const fn = x => void 0;"
`;

exports[`transform arrow functions expressions transforms correctly 1`] = `
"
const fn = x => x + 1;"
`;

exports[`transform arrow functions returns last directive if body is empty 1`] = `
"
const fn = x => {
    return \\"bar\\";
};"
`;

exports[`transform arrow functions works like do expressions 1`] = `
"
const fn = x => x > 0 ? \\"x>0\\" : \\"x<=0\\";"
`;
